quil-py/quil/program/__init__.pyi

```diff
@@ -120,6 +122,26 @@ class Program:
         Convert the instruction to a Quil string. If any part of the instruction can't
         be converted to valid Quil, it will be printed in a human-readable debug format.
         """
+    def wrap_in_loop(
+        self, loop_count_reference: MemoryReference, start_target: Target, end_target: Target, iterations: int
+    ) -> "Program":
+        """
+        Return a copy of the `Program` wrapped in a loop that repeats ``iterations`` times.
+
+        The loop is constructed by wrapping the body of the program in classical Quil instructions.
+        The given ``loop_count_reference`` must refer to an INTEGER memory region. The value at the
+        reference given will be set to ``iterations`` and decremented in the loop. The loop will
+        terminate when the reference reaches 0. For this reason your program should not itself
+        modify the value at the reference unless you intend to modify the remaining number of
+        iterations (i.e. to break the loop).
+
+        The given ``start_target`` and ``end_target`` will be used as the entry and exit points for
+        the loop, respectively. You should provide unique `quil.instructions.Target`s that won't be
+        used elsewhere in the program.
+
+        If `iterations` is 0, then a copy of the program is returned without any changes.
+        """
```

Comment:
Might also be worth noting that `iterations` must be non-negative

Reply:
Good call, added a note.